plugin/ProtocolHandler.pm

```diff
@@ -403,8 +403,66 @@ sub getNextTrack {
 
        		main::DEBUGLOG && $log->is_debug && $log->debug($http_result);
 
-       		my $streams = eval { decode_json($http_result) };
-       		my $streamInfo = getStreamJSON($streams->{'streamingData'}->{'adaptiveFormats'}, \@allowDASH);
+       		my $fields = eval { decode_json($http_result) };
+			my $status = $fields->{'playabilityStatus'}->{'status'};
+
+			if($status eq 'LOGIN_REQUIRED') {
+			    main::INFOLOG && $log->is_info && $log->info("Storing google visitor ID and trying again...");
+				$google_visitor = $fields->{'responseContext'}->{'visitorData'};
+				$cookieJar->set_cookie(0, 'GOOGLE_VISITOR', uri_escape($google_visitor), '/', '.youtube.com', undef, undef, 1, 3600*24*365);
+				Slim::Networking::SimpleAsyncHTTP->new(                                                                                                                                                                                                             sub {                                                                                                                       my $response = shift;                                                                                                   my $http_result = $response->content;                                                                                                                                                                                                           main::DEBUGLOG && $log->is_debug && $log->debug($http_result);                                                                                                                                                                                  my $fields = eval { decode_json($http_result) };
```

Comment:
Please make sure you fix the white space: use tabs to indent, rather than spaces. Add hard line endings where needed to make the code more human readable.

(that's just a general hint - I'm not the plugin maintainer, I didn't even look or even less understand the change...)